### PR TITLE
Fixed order of computeFields recomputations

### PR DESCRIPTION
**Pull request by bluenote10**
#124 seems to be caused by an issue I encountered recently. In the case of a segment overlap (`possibleIntersection` returns 2), the algorithm needs to re-run `computeFields`.

My understanding of this logic is: If there is an overlap with `next`, the sweep line order is:
```
next
event
prevEvent
```
In order to recompute the fields correctly from bottom to top, the recompute fields should be:
- recompute `event` based on `prevEvent`
- recompute `next` based on `event`

Basically the operands of the second call were swapped so far.

For comparison this is exactly how it is done in the overlap with `prev` case. Here the sweep line order is:
```
event
prevEvent
prevprevEvent
```
With the recompute fields call order:
- recompute `prevEvent` based on `prevprevEvent`
- recompute `event` based on `prevEvent`

The test case now works properly:

image

### CODE REVIEW

**Comment by w8r** [comment]
@bluenote10 thank you so much for this PR!
